File: NET-ReflectionFactoryMethod/VB/ExampleReflectionFactoryMethod/ScoreWithZero/Calculate.vb
Imports System.Collections.Generic
Imports IScore.Plugin

Namespace Score.Plugin
    Public Class Calculate
        Implements ICalculate

        Public Function GetScore(values As IList(Of Integer)) As Integer Implements ICalculate.GetScore
            If values Is Nothing OrElse values.Count = 0 Then
                Return 0
            End If

            Dim output As Integer = 0

            For Each item As Integer In values
                output += item
            Next

            Return output
        End Function
    End Class
End Namespace

File: NET-ReflectionFactoryMethod/CSharp/ExampleReflectionFactoryMethod/ScoreWithZero/Calculate.cs
using Score.Plugin.Interface;

namespace Score.Plugin
{
    public class Calculate : ICalculate
    {
        public int GetScore(IList<int> values)
        {
            int output = 0;

            foreach(var item in values)
            {
                output += item;
            }

            return output;
        }
    }
}
